kyo-zio/shared/src/main/scala/kyo/ZIOs.scala

```diff
@@ -0,0 +1,53 @@
+package kyo
+
+import ZIOs.internal.*
+import kyo.kernel.*
+import scala.util.control.NonFatal
+import scala.util.control.NoStackTrace
+import zio.Cause
+import zio.Task
+import zio.ZIO
+
+opaque type ZIOs <: Async = GetZIO & Async
+
+object ZIOs:
+
+    def get[E >: Nothing: Tag, A](v: ZIO[Any, E, A])(using Frame): A < (Abort[E] & ZIOs) =
+        val task = v.fold(Result.fail, Result.success)
+        Effect.suspendMap(Tag[GetZIO], task)(Abort.get(_))
+
+    def get[A](v: ZIO[Any, Nothing, A])(using Frame): A < ZIOs =
+        Effect.suspend(Tag[GetZIO], v)
+
+    inline def get[R: zio.Tag, E, A](v: ZIO[R, E, A])(using Tag[Env[R]], Frame): A < (Env[R] & ZIOs) =
+        compiletime.error("ZIO environments are not supported yet. Please handle them before calling this method.")
+
+    def run[E, A](v: A < (Abort[E] & ZIOs))(using Frame): ZIO[Any, E, A] =
+        ZIO.suspendSucceed {
```

Comment:
I think this can just become `suspend`:
https://github.com/zio/zio/blob/series/2.x/core/shared/src/main/scala/zio/ZIO.scala#L4803-L4813

Reply:
I had tried it initially but I get this error:

```
Found:    zio.ZIO[Any, E, A]
Required: zio.RIO[Nothing, Any]
```

It seems `ZIO.suspend` doesn't support typed errors in the produced computation.

Reply:
Ah right, this is because `RIO` is an alias for `ZIO[R, Throwable, A]`. In this case we need to support arbitrary `E` types. I think it would make sense to use `Exit.die` to propagate unexpected errors to ZIO's error channel.

Can you use `ZIO.fatalWith`?

Reply:
I imagine you mean `isFatalWith` to handle the exception? I've just updated it.